packages/notification-services-controller/src/NotificationServicesController/NotificationServicesController.ts

```diff
@@ -1136,15 +1147,14 @@ export default class NotificationServicesController extends BaseController<
       );
       const onChainNotifications = processAndFilter(rawOnChainNotifications);
 
-      const snapNotifications = this.state.metamaskNotificationsList.filter(
-        (notification) => notification.type === TRIGGER_TYPES.SNAP,
-      );
-
+      // Combine Notifications
```

Comment:
Added better comments to explain each step.

NOTE - this function does to a lot of things.
I think when we do the larger refactor we can think on how to isolate each notifications responsibilities.